src/App.tsx

```diff
@@ -10,21 +11,56 @@ import TableContainer from '@mui/material/TableContainer';
 import TableRow from '@mui/material/TableRow';
 import Paper from '@mui/material/Paper';
 import { TableHead } from "@mui/material";
+import { log_levels } from "./components/Log_Levels.tsx";
 
-type getMessageReturn = [string[],string[],string[],string[],number[],string[]]
+class MessageReturn {
+  timestamp: string[] = [];
+  host: string[] = [];
+  log_level_str: string[] = [];
+  log_message: string[] = [];
+  log_level: number[] = [];
+  app_name: string[] = [];
+}
```

Comment:
You now have an excellent scope for simplifying the code:

What you're doing here is downloading a row-wise table from the server (which comes out as a list of `LogData`s) and then using a bunch of code to transpose into a column-wise table (a `MessageReturn` with lists of each category) and then applying the column-wise table to the DOM.

You can just cut out the middle man and have the state as a list of records, which would greatly simplify `getMessage()`. I would suggest something like this:

```suggestion
interface LogRecord {
  timestamp: string;
  host: string;
  log_level_str: string;
  log_message: string;
  log_level: number;
  app_name: string;
}
```

And then the state is

```python
const [LogResponse,setlogResponse] = useState<[LogRecord]>([]);
```

If you try that and refactor everything else around it I think the code will come out much simpler.